controllers/platformnetwork_controller.go

```diff
@@ -661,37 +698,220 @@ func (r *PlatformNetworkReconciler) statusUpdateRequired(instance *starlingxv1.P
 	return !instance.Status.DeepEqual(status)
 }
 
-// ReconcileResource interacts with the system API in order to reconcile the
-// state of a data network with the state stored in the k8s database.
-func (r *PlatformNetworkReconciler) ReconcileResource(client *gophercloud.ServiceClient, instance *starlingxv1.PlatformNetwork) (err error) {
-	oldStatus := instance.Status.DeepCopy()
+// Determine if the system type is All-in-one simplex, return the Host object if so.
+func (r *PlatformNetworkReconciler) AIOSXHost(client *gophercloud.ServiceClient) (*hosts.Host, bool, error) {
+	all_hosts, err := hosts.ListHosts(client)
+	if err != nil {
+		return nil, false, common.NewSystemDependency("Failed to get host list")
+	}
 
-	if instance.DeletionTimestamp.IsZero() {
-		err = r.ReconcileAddressPool(client, instance)
-		if err == nil {
-			if networkResourceRequired(instance) {
-				err = r.ReconcileNetwork(client, instance)
+	if len(all_hosts) == 1 {
+		for _, host := range all_hosts {
+			if host.Personality == "controller" {
+				default_system, err := system.GetDefaultSystem(client)
+				if err != nil {
+					return nil, false, common.NewSystemDependency("Failed to get default system")
+				}
+
+				if default_system.SystemMode == "simplex" && default_system.SystemType == "All-in-one" {
```

Comment:
why are you not using this to check? Also use consts.

Reply:
Considering SX -> DX upgrade or other such scenarios leading to propagation of stale entries we are using two other conditions `if len(all_hosts) == 1` and `if host.Personality == cloudManager.PersonalityController` to be on the safer side.
Updated to use consts as suggested but keeping the conditions for now, if you still think we can manage it just the `default_system.SystemMode` and `default_system.SystemType` do let me know, I'll remove it.